### PR TITLE
MARP-228-Problem-with-meta-proc-connector

### PR DESCRIPTION
**Pull request by tutn-axonivy**

### CODE REVIEW

**Comment by tutn-axonivy** [comment]
Hi @ivy-rew, I create this pull to update as your feedback. Could you review it? Thanks.

**Comment by ivy-rew** [comment]
> Hi @ivy-rew, I create this pull to update as your feedback. Could you review it? Thanks.

as discussed in the chat: I can do a review. but first do the internal review.

File: metaproc-connector-demo/config/variables.yaml [hunk 1]
```diff
@@ -3,4 +3,4 @@ Variables:
     Url: http://localhost:8080
     Username: admin
     Password: 1234
-    TokenExpiredDuration: 3600000 # Milisecond
+    TokenExpiredDuration: 60 # Minute
```
Comment (ghost) [inline metaproc-connector-demo/config/variables.yaml]: We should add a comment above the variable according to Reguel's suggestion.
"# Duration in minutes
TokenExpiredDuration: 60"

File: metaproc-connector/src/com/axonivy/connector/metaproc/auth/AuthTokenFeature.java [hunk 2]
```diff
@@ -53,11 +53,13 @@ public void filter(ClientRequestContext context) throws IOException {
...
-        TOKEN_EXPIRED_TIME = System.currentTimeMillis() + Long.valueOf(Ivy.var().get("metaproc.TokenExpiredDuration"));
+        TOKEN_EXPIRED_TIME = System.currentTimeMillis() + getExpiredTime();
         context.getHeaders().add(TOKEN_HEADER, TOKEN);
-      } catch (Exception e) {
-        Ivy.log().error(e.getMessage());
       }
     }
   }
+  
+  private long getExpiredTime() {
```
Comment (ghost) [inline metaproc-connector/src/com/axonivy/connector/metaproc/auth/AuthTokenFeature.java]: rename to getExpiredDuration

File: metaproc-connector-demo/config/variables.yaml [hunk 3]
```diff
@@ -3,4 +3,5 @@ Variables:
     Url: http://localhost:8080
     Username: admin
     Password: 1234
-    TokenExpiredDuration: 60 # Minute
+    # Token expiration in minute
```
Comment (ivy-rew) [inline metaproc-connector-demo/config/variables.yaml]: typo: minute -> minutes

File: metaproc-connector/src/com/axonivy/connector/metaproc/auth/AuthTokenFeature.java [hunk 4]
```diff
@@ -53,13 +53,13 @@ public void filter(ClientRequestContext context) throws IOException {
...
-        TOKEN_EXPIRED_TIME = System.currentTimeMillis() + getExpiredTime();
+        TOKEN_EXPIRED_TIME = System.currentTimeMillis() + getExpiredDuration();
         context.getHeaders().add(TOKEN_HEADER, TOKEN);
       }
     }
   }
   
-  private long getExpiredTime() {
+  private long getExpiredDuration() {
 	  return Long.valueOf(Ivy.var().get("metaproc.TokenExpiredDuration")) * 60 * 1000;
```
Comment (ivy-rew) [inline metaproc-connector/src/com/axonivy/connector/metaproc/auth/AuthTokenFeature.java]: time conversions are hard to read and maintain: it's better to convert em using java's time helpers: 
```
var minutes = Long.valueOf(Ivy.var().get("metaproc.TokenExpiredDuration"));
return TimeUnit.MINUTES.toMillis(seconds);
```

File: metaproc-connector-demo/config/variables.yaml [hunk 5]
```diff
@@ -3,4 +3,5 @@ Variables:
...
+    TokenExpiredDuration: 60
```
Comment (ivy-rew) [inline metaproc-connector-demo/config/variables.yaml:7]: is 60 minutes a default we know from meta-proc? or do we expect all customers to customize this? maybe @thomaszehetner has some insights here?

Comment (tutn-axonivy) [reply metaproc-connector-demo/config/variables.yaml:7]: I don't know exactly what type of token this API uses. So I made it editable for manual expire logic.

**Comment by thomaszehetner** [comment]
feedback from meta:proc: default token expiration is 60 minutes

File: metaproc-connector-demo/config/variables.yaml (hunk 5 above)
Comment (thomaszehetner) [reply metaproc-connector-demo/config/variables.yaml:7]: feedback from meta:proc: default token expiration time is 60 minutes

Comment (ivy-rew) [reply metaproc-connector-demo/config/variables.yaml:7]: Thanks a lot for your feedback @thomaszehetner. So, let's keep this default.

**Review by ivy-rew — APPROVED** [review]
Thanks

**Pull request merged**